store.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ratify
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/registry"
+)
+
+// registeredStores saves the registered store factories.
+var registeredStores map[string]func(config CreateStorOptions) (Store, error)
+
+// Store is an interface that defines methods to query the graph of supply chain content including its related content
+type Store interface {
+	// Name is the name of the store
+	Name() string
+
+	// ListReferrers returns the immediate set of supply chain artifacts for the given subject
+	// represented as artifact manifests.
+	// Note: This API supports pagination. fn should be set to handle the underlying pagination.
+	ListReferrers(ctx context.Context, subjectReference registry.Reference, subjectDescriptor ocispec.Descriptor, artifactTypes []string, fn func(referrers []ocispec.Descriptor) error) ([]ocispec.Descriptor, error)
+
+	// GetBlobContent returns the blob with the given digest and its subject.
+	// WARNING: This API is intended to use for small objects like signatures, SBoMs.
+	GetBlobContent(ctx context.Context, subjectReference string, digest digest.Digest) ([]byte, error)
+
+	// GetReferenceManifest returns the referenced artifact manifest as given by the descriptor and its subject.
+	GetReferenceManifest(ctx context.Context, subjectReference string, referenceDesc ocispec.Descriptor) (ocispec.Manifest, error)
```

Comment:
Why we need to provide `referenceDesc` as we have `subjectReference`? Is it intended as you want to verify the size and the media type? If so, `referenceDesc` should be named as `expected`.